### ISSUE TITLE
Can't distinguish a reestablish with a previously `leave` called and a simple reconnection

### ISSUE THREAD

**Issue by davidyuk** [aeternity/aeternity#4399]
There are two flows when a peer reconnects to a previously established state channel:
- with `leave` called before,
- without `leave` called before.

In both cases, the client needs to provide `existingChannelId`, `existingFsmId`.

After opening the WS connection the first message in both cases is
```js
{
  method: 'channels.info',
  params: {
    channel_id: 'ch_FumbzCRZrSv9omagbmur9hvWNnpfj2qnSRdz3HBB7zZs7vQ8c',
    data: {
      event: 'fsm_up',
      fsm_id: 'ba_JVLBKiw38bIm9WmmMD9+EJ3mzFhJrWeRPyw7E4dSHO7uDbK/'
    }
  }
}
```
after that in the 2nd case, the connection is ready to use and no extra messages are received. But in the first case, extra messages are received:
```js
{
  method: 'channels.info',
  params: {
    channel_id: 'ch_FumbzCRZrSv9omagbmur9hvWNnpfj2qnSRdz3HBB7zZs7vQ8c',
    data: { event: 'channel_reestablished' }
  }
}
{
  method: 'channels.info',
  params: {
    channel_id: 'ch_FumbzCRZrSv9omagbmur9hvWNnpfj2qnSRdz3HBB7zZs7vQ8c',
    data: { event: 'open' }
  }
}
{
  method: 'channels.update',
  params: {
    channel_id: 'ch_FumbzCRZrSv9omagbmur9hvWNnpfj2qnSRdz3HBB7zZs7vQ8c',
    data: {
      state: 'tx_+NILAfiEuEAKT+XRGctLvkEB6C36yXnSLfBQf1L5K/hKHS1lgAu/nZheHDrYudIG4jasapGnUT7UnuLstjJ/esPwbPNxZHcFuEAPteRwcDzqx+B3fQ6ajye4dpBzoLnP33zZwUfe6vgpWRx/AeV+urbVu50dMIMAWr2qID8TjDGLaKSZOJUAnlcHuEj4RjkCoQYh2sO79BldQP1eSs4GEFw9hz/P3RIomAVPdKyXeaCwzAKg+hVWSvB0+frb8i4kbYTbvWubjLRHN+CX5TlfQme4Rm0FDqdH'
    }
  }
}
```
The problem is there is no indication of which flow is it, so the developer needs to remember it and provide a corresponding flag to JS SDK in connection parameters.

Otherwise, SDK can't detect if the connection is opened or is still opening. Probably an extra `{ event: 'open' }` in the 2dn flow would help.

**Comment by hanssv** [comment]
Well, in case 1 (with a `leave`) you are obliged to keep track of the state yourself, in case 2 (no `leave`) you'll get the state from the FSM, or did I misunderstand the scenario? I.e. the developer _must_ remember or else he might end up in a situation where he doesn't know the state?!

**Comment by davidyuk** [comment]
> in case 1 you are obliged to keep track of the state yourself

Not really, the state is provided by the node on connection. And in the 2 case, a peer can request it manually.
As I understand in both cases peer can get current state from the node, and it both cases only `channelId`, `fsmId` are required to reconnect.

**Comment by hanssv** [comment]
I don't think I understand the use-case/scenario, but regardless, if the server is stateful and behaves differently depending on the state it sounds like a good idea to expose this state to the user! 👍

**Comment by davidyuk** [comment]
Maybe we don't need `leave` method at all 🤷‍♀️  to sent a message to a peer if another peer disconnects or ping fails.
